distributed/utils.py

```diff
@@ -335,6 +335,7 @@ def sync(loop, func, *args, callback_timeout=None, **kwargs):
     main_tid = threading.get_ident()
     result = [None]
     error = [False]
+    _future = [None]
```

Comment:
you don't need to use this list-cell-pattern with results in python3+, you can use `nonlocal _future` in `def f():`

Reply:
Thanks.  I cleaned this up a bit.  I think I just set a pattern many years ago and kept with it :)

Reply:
Much cleaner now :)